src/providers/FusionAuthProvider.tsx

```diff
@@ -182,35 +181,48 @@ export const FusionAuthProvider: React.FC<FusionAuthConfig> = props => {
 
     const setUserFromCookie = useCallback((cookie: string) => {
         try {
+            /* JSON parse needs try/catch to not crash app */
             const parsedUserCookie = JSON.parse(cookie);
             setUser(parsedUserCookie);
         } catch {
-            /* if JSON parse fails doesn't crash the app */
+            setUser({});
+            Cookies.remove('user');
         }
     }, []);
 
     const fetchUserFromServer = useCallback(async () => {
         setIsLoading(true);
+        // lastState indicates that this is a redirect
+        const lastState = Cookies.get('lastState');
+
         try {
             const response = await fetch(
                 generateServerUrl(ServerFunctionType.me, mePath),
                 {
                     credentials: 'include',
                 },
             );
+
+            if (!response.ok) {
+                throw response;
```

Comment:
Typically we want to throw an `Error` instead of arbitrary data types since that's what is usually expected.

```ts
throw new Error(`Unable to fetch user. Request failed with status code ${response.status}`)
```

The associated `catch` block takes in a `Error` at the moment as well, so throwing `response` is a notable change from what `onRedirectFail` has been taking in so far.